go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -878,3 +916,55 @@ func (bsnc *BaseSecondaryNetworkController) requireDHCP(pod *corev1.Pod) bool {
 		bsnc.IsPrimaryNetwork() &&
 		bsnc.TopologyType() == types.Layer2Topology
 }
+
+func (bsnc *BaseSecondaryNetworkController) setPodLogicalSwitchPortEnabledField(
+	pod *corev1.Pod, nadName string, ops []ovsdb.Operation, enabled bool) ([]ovsdb.Operation, *nbdb.LogicalSwitchPort, error) {
+	lsp := &nbdb.LogicalSwitchPort{Name: bsnc.GetLogicalPortName(pod, nadName)}
+	lsp, err := libovsdbops.GetLogicalSwitchPort(bsnc.nbClient, lsp)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to fetch switch port for node %s: %w", lsp.Name, err)
+	}
+
+	lsp.Enabled = ptr.To(enabled)
+	switchName, err := bsnc.getExpectedSwitchName(pod)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to fetch switch name for pod %s: %w", pod.Name, err)
+	}
+	customFields := []libovsdbops.ModelUpdateField{libovsdbops.LogicalSwitchPortEnabled}
+	ops, err = libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitchWithCustomFieldsOps(bsnc.nbClient, ops, &nbdb.LogicalSwitch{Name: switchName}, customFields, lsp)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed updating logical switch port %+v on switch %s: %w", *lsp, switchName, err)
+	}
+	return ops, lsp, nil
```

Comment:
- Why are you doing the `GetLogicalSwitchPort`?
- Are we sure we can't have a race on the DB data that could happen if the node that owns the source LSP is making updates on it at the same time?

Reply:
Maybe this should just be an update and not a create? As long as we are sure the the node that own the LSP can't possibly be updating LSP.Enabled

Reply:
> * Why are you doing the `GetLogicalSwitchPort`?

The solution of setting the LSP of the target,source pods is triggered on the target pod. This means that the pod reconciled (and its LSP) is of the target-pod. 
This function however is meant to change the LSP of the *source* pod, i.e. not the one that is currently reconciled. This is why I need to get it.

> * Are we sure we can't have a race on the DB data that could happen if the node that owns the source LSP is making updates on it at the same time?
That could potentially happen, but since we're only changing the LSP.Enabled field, and we are the only ones that change it - then there is no actual race.

Reply:
> Maybe this should just be an update and not a create? As long as we are sure the the node that own the LSP can't possibly be updating LSP.Enabled

[DONE](https://github.com/ovn-kubernetes/ovn-kubernetes/compare/b95c7de049a65c228fa711a9630f49f339e80171..dcac63f80853b19d2dfd207de59f66ae220d44cd)

Reply:
> The solution of setting the LSP of the target,source pods is triggered on the target pod. This means that the pod reconciled (and its LSP) is of the target-pod.
> This function however is meant to change the LSP of the _source_ pod, i.e. not the one that is currently reconciled. This is why I need to get it.

I still don't see why you have to get it. You are not reading anything from it.

> That could potentially happen, but since we're only changing the LSP.Enabled field, and we are the only ones that change it (by setting the Enabled field only when necessary) - then there is no actual race.

Except that you might be creating the LSP (`CreateOrUpdate`) and, if the owner node has already deleted it, the you are creating an LSP that nobody is going to ever delete?
Also, could it happen that the source LSP is already that from a migrated/migrating VM where the owner node may also be playing with the Enabled field?

Reply:
OK I see that you added the update. However it will fail if the LSP does not exist. Just crosschecking with you if that's what you want to happen or if you should ignore the NotFound error.

Reply:
> > The solution of setting the LSP of the target,source pods is triggered on the target pod. This means that the pod reconciled (and its LSP) is of the target-pod.
> > This function however is meant to change the LSP of the _source_ pod, i.e. not the one that is currently reconciled. This is why I need to get it.
> 
> I still don't see why you have to get it. You are not reading anything from it.

ah I see now, my bad. Removing.

> 
> > That could potentially happen, but since we're only changing the LSP.Enabled field, and we are the only ones that change it (by setting the Enabled field only when necessary) - then there is no actual race.
> 
> Except that you might be creating the LSP (`CreateOrUpdate`) and, if the owner node has already deleted it, the you are creating an LSP that nobody is going to ever delete? 

This is solved by your suggestion to do Update and not CreateOrUpdate.

> Also, could it happen that the source LSP is already that from a migrated/migrating VM where the owner node may also be playing with the Enabled field?

That is not possible AFAIK. @qinqon keep me honest here

Reply:
@RamLavi @jcaamano this is the only place where we set the enabled field so there is no race changing it's value I think, also the virt-launcher pods are new everytime so LSPs are new everytime, so we cannot be playing from LSPs from another migration.